libcxx/include/__utility/pair.h

```diff
@@ -284,7 +284,8 @@ struct _LIBCPP_TEMPLATE_VIS pair
     }
 
     template <__pair_like _PairLike>
-      requires(is_constructible_v<first_type, decltype(std::get<0>(std::declval<_PairLike&&>()))> &&
+      requires(!__is_specialization_of_subrange<remove_cvref_t<_PairLike>>::value &&
```

Comment:
Is this change tested?

Reply:
I am bit struggled to test this.  It turns out that `subrange` has `operator PairLike`

Even with this constructor removed, the
```
std::pair p(subrange)
```
still works

i.e. with and without this constructor, we can always construct a `pair` from a `subrange`

This test still passes on my side after the change
[LWG 3525](https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/utilities/utility/pairs/pairs.pair/ctor.pair_like.pass.cpp#L85)

any idea of how to test this?

Reply:
Actually I found an example where without the fix, it is ambiguous between `pair`'s constructor and `subranges`' `operator Pairlike`.

https://godbolt.org/z/fefaEzz9M